client/src/lib/data.ts

```
@@ -30,9 +30,30 @@ export type TripEntry = {
   iconUrl: string;
 };
 
-export async function getEvents(userId: number) {
+export const placeholder: TripEntry = {
+  tripId: 1,
+  userId: 1,
+  tripName: '',
+  startDate: new Date(),
+  endDate: new Date(),
+  iconUrl: 'placeholder-image',
+};
+
+export async function getAllTrips(userId: number) {
+  try {
```

You don't want try/catch here since then there will be no way to inform the user that an error occurred. You want this function to let the exception propagate up to the caller, which should always have a try/catch, where an error can be displayed to the user.

Look at what you are doing in the catch. You just log it to the console. Users don't see the console, so they will never see this error. Then after the catch you return, and since there's nothing to return, you will return `undefined`. The caller will then likely not know what to do with undefined and will throw its own Error!

This applies to all your functions in here that call fetch.